Assets/HoloToolkit/Sharing/Scripts/SDK/ObjectElementAdapter.cs

```diff
@@ -11,61 +11,160 @@ namespace HoloToolkit.Sharing
     public class ObjectElementAdapter : ObjectElementListener
     {
         public event System.Action<long, int> IntChangedEvent;
+        public event System.Action<long, double> DoubleChangedEvent;
         public event System.Action<long, float> FloatChangedEvent;
+        public event System.Action<long, long> LongChangedEvent;
         public event System.Action<long, XString> StringChangedEvent;
+        public event System.Action<long, bool> BoolChangedEvent;
         public event System.Action<Element> ElementAddedEvent;
         public event System.Action<Element> ElementDeletedEvent;
 
-        public ObjectElementAdapter() { }
+        /// <summary>
+        /// The Object Elemet we're listening to.
+        /// </summary>
+        public ObjectElement ObjectElementRoot { get; private set; }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="ObjectElementAdapter"/> that creates a new Object Element and starts to listen to it.
+        /// </summary>
+        public ObjectElementAdapter()
```

Comment:
I'm not sure if creating a new Object Element in the constructor is a good idea, but it seems to be the pattern in other test apps.  I'd like to get more feedback on this, if possible.